src/agent_sdk/inc/du_agent_sdk/agent_module_interface.h

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file agent_module_interface.h
+ * @brief Defines types and methods for DU Agent module library.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#ifndef ADUC_AGENT_MODULE_INTERFACE_H
+#define ADUC_AGENT_MODULE_INTERFACE_H
+
+#include "du_agent_sdk/agent_common.h"
+#include "aduc/result.h"
+
+#ifdef __cplusplus
```

Comment:
already #include'ing agent_common.h above, so might as well use EXTERN_C_BEGIN and EXTERN_C_END in here instead of duplicating.